src/components/AddPlaidBankAccount.js

```diff
@@ -78,9 +78,11 @@ class AddPlaidBankAccount extends React.Component {
 
     componentDidMount() {
         // If we're coming from Plaid OAuth flow then we need to reuse the existing plaidLinkToken
+        // If there is already a selected Plaid accountID then we don't need to trigger the Plaid flow
         if ((this.props.receivedRedirectURI && this.props.plaidLinkOAuthToken)
             || !_.isEmpty(lodashGet(this.props.plaidData, 'bankAccounts'))
-            || !_.isEmpty(lodashGet(this.props.plaidData, 'errors'))) {
+            || !_.isEmpty(lodashGet(this.props.plaidData, 'errors'))
+            || !_.isEmpty(this.props.selectedPlaidAccountID)) {
```

Comment:
Hmmm do you know why we're re-entering this view in the first place? I wonder if the fix shouldn't happen outside of the view so this component doesn't get called.

Reply:
For the issue https://github.com/Expensify/App/issues/15197 we are entering via deep-linking, directly using the URL so it makes sense. 

But for another issue, I am confused too.

Reply:
> Hmmm do you know why we're re-entering this view in the first place?

My theory is that when the user clicks the `Continue` button on the [PBA success page](https://github.com/Expensify/App/blob/maria-fix-plaid-retrigger/src/pages/AddPersonalBankAccountPage.js#L95-L105), because we clear `ONYXKEYS.PERSONAL_BANK_ACCOUNT`, [shouldShowSuccess](https://github.com/Expensify/App/blob/347a4f9e51d81285473a850e118b0f7ac619f671/src/pages/AddPersonalBankAccountPage.js#L85) evaluates to false and the [AddPlaidBankAccount](https://github.com/Expensify/App/blob/347a4f9e51d81285473a850e118b0f7ac619f671/src/pages/AddPersonalBankAccountPage.js#L106-L127) component is mounted again.

But I'm not 100% why it only happens during wallet KYC and via deep-linking.

Reply:
Let me know what you think, @nkuoch @Santhosh-Sellavel

Reply:
Why do we need to clear details here?https://github.com/Expensify/App/blob/347a4f9e51d81285473a850e118b0f7ac619f671/src/pages/AddPersonalBankAccountPage.js#L101-L103

Does it affect anything else, if it needs to be cleared we can do it at `componentWillUnmount`

@MariaHCD or @nkuoch

Reply:
Good idea. I think we can remove `BankAccounts.clearPersonalBankAccount();` from onButtonPress. The original idea was to make sure it's cleared for any next setup, but we already call it in componentDidMount, so it's probably useless here.

Reply:
Oh, nice catch, @Santhosh-Sellavel! I think it would make sense to call it in `componentWillUnmount`. We're also calling it at [componentDidMount](https://github.com/Expensify/App/blob/main/src/pages/AddPersonalBankAccountPage.js#L65-L67) but maybe it makes more sense to just have it in `componentWillUnmount` just so that there isn't any outdated data in the Onyx keys once the PBA is added. Thoughts?

Reply:
I agree let me know when changes are done!